### ISSUE TITLE
Matching of sending IP address to find pod info should strip the port number first

### ISSUE THREAD

**Issue by jphx** [grafana/agent#1984]
I'm using the Grafana Agent, version 0.26.1, configured using the tk/jsonnet configuration alternative.  My configuration file includes this to enable translation of the sending IP address into pod information:

```yaml
scrape_configs:
  - bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
    job_name: kubernetes-pods
    kubernetes_sd_configs:
      - role: pod
    relabel_configs:
      - action: replace
        source_labels:
          - __meta_kubernetes_namespace
        target_label: namespace
      - action: replace
        source_labels:
          - __meta_kubernetes_pod_name
        target_label: pod
      - action: replace
        source_labels:
          - __meta_kubernetes_pod_container_name
        target_label: container
    tls_config:
      ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
      insecure_skip_verify: false
```

However, I see no pod information in the captured traces.  After enabling debug-level logging in the Grafana Agent, I see this error message when the trace is received:

```
ts=2022-08-03T13:07:49.133188793Z caller=prom_sd_processor.go:166 level=debug component="traces service disco" msg="unable to find matching hostLabels" ip=172.30.114.52:54443
```

I believe the Grafana Agent knows about the IP address because I see messages like this earlier in the log:
```
ts=2022-08-03T13:07:26.142710099Z caller=prom_sd_processor.go:284 level=debug component="traces service disco" msg="adding host to hostLabels" host=172.30.114.52
```

I believe the presence of the port number in the string being looked up is causing the failure to find the match.  The port number should be stripped before doing the lookup.

**Comment by rfratto** [comment]
:wave:

There's a few ways the IP gets retrieved. The [docs][] mention that by default, it's looking at a few labels on the span to find the IP of the sender:

```
# Options are `ip`, `net.host.ip`, `k8s.pod.ip`, `hostname` and `connection`.
#   - `ip`, `net.host.ip` and `k8s.pod.ip`, `hostname` match spans tags.
#   - `connection` inspects the context from the incoming requests (gRPC and HTTP).
```

I'll look at each of those in order by default; you can change the value with the `prom_sd_pod_associations` field.

So the port number is coming from one of your labels on your spans. The port number does get stripped from the service discovery, but not the value from the span label it finds to associate with.

[docs]: https://grafana.com/docs/agent/latest/configuration/traces-config/

**Comment by jphx** [comment]
If it helps in the research, this span was delivered using the OpenCensus protocol.

**Comment by jphx** [comment]
I'm new to traces, but I looked through the "Tags" and "Process" sections of the span and don't see the sending IP address in there at all.  Probably this means the IP address (and port) were retrieved using the `connection` method?  I tried looking through the code, but in the time I spent, I am unable to to determine where the `Info.Addr` field gets set.  However, the field is of type `net.Addr`, which I believe can have the port number in the string.
